File: blastoff_app/app/schemas/rocket.py
from typing import Optional

from pydantic import BaseModel, Field


class RocketBase(BaseModel):
    name: str
    no_stages: Optional[int] = Field(
        ..., gt=0, description="Number of stages, must be a positive integer"
    )
    height: Optional[float] = None
    mass: Optional[float] = None
    diameter: Optional[float] = None
    description: Optional[str] = None
    launches_count: Optional[int] = Field(
        0, ge=0, description="Total number of launches"
    )
    successful_launches_count: Optional[int] = Field(
        0, ge=0, description="Number of successful launches"
    )
    failed_launches_count: Optional[int] = Field(
        0, ge=0, description="Number of failed launches"
    )
    landings_count: Optional[int] = Field(
        0, ge=0, description="Total number of landings"
    )
    successful_landings_count: Optional[int] = Field(
        0, ge=0, description="Number of successful landings"
    )
    failed_landings_count: Optional[int] = Field(
        0, ge=0, description="Number of failed landings"
    )
    pending_launches: Optional[int] = Field(
        0, ge=0, description="Total number of pending launches"
    )
    leo_capacity: Optional[float] = None
    gto_capacity: Optional[float] = None
    geo_capacity: Optional[float] = None
    sso_capacity: Optional[float] = None
    rocket_thrust: Optional[float] = None
    launch_cost: Optional[float] = None
    image_url: Optional[str] = None


class RocketCreate(RocketBase):
    id: int
    agency_id: int


class RocketResponse(RocketBase):
    id: int
    agency_id: int

    class Config:
        from_attributes = True

File: mobile/features/rockets/shemas/rocket.ts
import z from "zod";

export const Agency = z.object({
  id: z.number(),
  name: z.string().nullable(),
  country: z.string().nullable(),
  description: z.string().nullable(),
  website: z.string().nullable(),
  image_url: z.string().nullable(),
});

export type Agency = z.infer<typeof Agency>;

export const Rocket = z.object({
  id: z.number(),
  name: z.string(),
  description: z.string(),
  image_url: z.string().nullable(),
  agency: Agency,
  diameter: z.number().nullable(),
  height: z.number().nullable(),
  no_stages: z.number().nullable(),
  mass: z.number().nullable(),
  rocket_thrust: z.number().nullable(),
  leo_capacity: z.number().nullable(),
  gto_capacity: z.number().nullable(),
  geo_capacity: z.number().nullable(),
  sso_capacity: z.number().nullable(),
  launch_cost: z.number().nullable(),
  launches_count: z.number().nullable(),
  successful_launches_count: z.number().nullable(),
  failed_launches_count: z.number().nullable(),
  pending_launches: z.number().nullable(),
  landings_count: z.number().nullable(),
  successful_landings_count: z.number().nullable(),
  failed_landings_count: z.number().nullable(),
});

export type Rocket = z.infer<typeof Rocket>;
